javascripts/discourse/components/category-list.js

```diff
@@ -0,0 +1,19 @@
+import Category from "discourse/models/category";
+import Component from "@glimmer/component";
+import I18n from "I18n";
+
+export default class CategoryList extends Component {
+  constructor() {
+    super(...arguments);
+    this.blockTitle =
+      this.args?.params?.title || I18n.t(themePrefix("category_list.heading"));
+
+    this.categoryList = this.args?.params?.id.split(",");
+    this.categoryList.forEach(function (id, i, arr) {
+      arr[i] = Category.findById(id);
+    });
+    if (!this.categoryList) {
```

Comment:
This needs to be before the `forEach` loop, essentially it protects from the loop failing if `this.categoryList` is empty.

Reply:
Ah okay, thank you for the explanation. Does this need to be done for [tag-topics.js](https://github.com/discourse/discourse-right-sidebar-blocks/blob/main/javascripts/discourse/components/tag-topics.js) and [top-topics.js](https://github.com/discourse/discourse-right-sidebar-blocks/blob/main/javascripts/discourse/components/top-topics.js) as well?